repository/src/main/java/org/alfresco/repo/search/impl/querymodel/impl/db/DBQueryEngine.java

```diff
@@ -335,16 +342,19 @@ public void handleResult(ResultContext<? extends Node> context)
             private void doHandleResult(NodePermissionAssessor permissionAssessor, List<Node> nodes,
                                         int requiredNodes, ResultContext<? extends Node> context)
             {
-                if (nodes.size() >= requiredNodes)
+                if (!maxPermissionCheckEnabled)
                 {
-                    context.stop();
-                    return;
+                    if (nodes.size() >= requiredNodes)
+                    {
+                        context.stop();
+                        return;
+                    }
                 }
```

Comment:
To avoid curly brackets proliferation I would probably rewrite this as:

```
                if (!maxPermissionCheckEnabled && nodes.size() >= requiredNodes)                    
                {                        
                    context.stop();                        
                    return;                    
                }
```

Reply:
Oh yeah good catch!